### ISSUE TITLE
Publisher cannot update api definitions from resources page

### ISSUE THREAD

**Issue by ecostanzi** [wso2/product-apim#7601]
### Description:
I cannot update the api definitions from the Resources page of the API details. I was trying to disable security for one openapi endpoint and this error showed up. Then I noticed that the error appears even without touching the resources configuration.

### Steps to reproduce:
- create a new api using https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v3.0/petstore.yaml
- open the API details and click on the "Resources" tab
- scroll to the bottom of the Resources page and click on "Save"

The following error appears:
```
[900754]: Error while parsing OpenAPI definition
attribute components.schemas.Pets.$$ref is unexpected
```

### Affected Product Version:

WSO2 API-M v3.0.0 

### Environment details (with versions):
- OS: Windows 10 Professional
- Client: Chrome
- Env (Docker/K8s): Standalone installation on Windows 10 Pro

---

**Comment by Krishanx92** [comment]
Thanks for pointing this out. We have identified this and fixed it in the upcoming 3.1.0 version.
Refer: https://github.com/wso2/product-apim/issues/6597

**Comment by ecostanzi** [comment]
Ok thanks, sorry for the duplicate issue
